compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -166,6 +168,12 @@ class ExpressionCodegen(
 
     fun markLineNumber(element: IrElement) = element.markLineNumber(true)
 
+    fun noLineNumberScope(block: () -> Unit) {
+        noLineNumberScope = true
+        block()
+        noLineNumberScope = false
```

Comment:
I would suggest to keep old state and restore it here to avoid bugs in future

Reply:
Done.

Reply:
Thank you! That was last from my side, but there is one more from TC:  failed SerializationIrBytecodeListingTestGenerated.testBasic test

Reply:
There is a dump of the full bytecode for the IR backend. The only change is the removal of the end brace line numbers. The actual code is the same. Rebased so the test passes. As far as I can tell, the point of the test is to check that serialization code is correctly generated.